pkg/common/timeutils/timeutils.go

```diff
@@ -78,41 +79,44 @@ func GetStartEndTime(params GetStartEndTimeParams) (*time.Time, *time.Time, erro
 		return nil, nil, fmt.Errorf("could not parse provided timeframe: %s", err.Error())
 	}
 
-	// initialize default values for end and start time
-	end := time.Now().UTC()
-	start := time.Now().UTC().Add(-timeframeDuration)
+	// calculate possible start and end time
+	now := time.Now().UTC()
+	calcStart := now.Add(-timeframeDuration)
+
+	// initialize default values for end and start time pointer
+	var end, start *time.Time
+	end = &now
+	start = &calcStart
```

Comment:
I think the previous code it is easier to read, but I would have it with some of your changes
```suggestion
	end := time.Now().UTC()
	start := end.Add(-timeframeDuration)
```